inst/include/population_dynamics/recruitment/functors/recruitment_base.hpp

```diff
@@ -18,106 +18,157 @@
 
 #include "../../../common/fims_math.hpp"  // for using fims::log()
 #include "../../../common/model_object.hpp"
+#include "../../../distributions/distributions.hpp"
 
 namespace fims {
 
-/** @brief Base class for all recruitment functors.
- *
- * @tparam Type The type of the recruitment functor.
- *
- */
-template <class Type>
-struct RecruitmentBase : public FIMSObject<Type> {
-  static uint32_t id_g; /*!< reference id for recruitment object*/
-
-  std::vector<Type>
-      recruit_deviations;           /*!< A vector of recruitment deviations */
-  bool constrain_deviations = true; /*!< A flag to indicate if recruitment
+    /** @brief Base class for all recruitment functors.
+     *
+     * @tparam Type The type of the recruitment functor.
+     *
+     */
+    template <class Type>
+    struct RecruitmentBase : public FIMSObject<Type> {
+        static uint32_t id_g; /*!< reference id for recruitment object*/
+
+        typename ModelTraits<Type>::ParameterVector
+          recruit_deviations; /*!< A vector of recruitment deviations */
+        bool constrain_deviations = false; /*!< A flag to indicate if recruitment
                                        deviations are summing to zero or not */
-  std::vector<Type> recruit_bias_adjustment; /*!< A vector of bias adj values
+        typename ModelTraits<Type>::DataVector recruit_bias_adjustment; /*!< A vector of bias adj values
                                                 (incorporating sigma_recruit)*/
-  // Initially fixing bias adjustment (b_y in collobarative
-  // workflow specification) to 1.0.
-  // In the future, this would be set by the user.
-  std::vector<Type>
-      recruit_bias_adjustment_fraction; /*!< A vector of bias adjustment
+        // Initially fixing bias adjustment (b_y in collobarative
+        // workflow specification) to 1.0.
+        // In the future, this would be set by the user.
+        typename ModelTraits<Type>::DataVector
+        recruit_bias_adjustment_fraction; /*!< A vector of bias adjustment
                                            fractions (on the 0 to 1 range)*/
-  bool use_recruit_bias_adjustment =
-      true; /*!< A flag to indicate if recruitment deviations are bias adjusted
+        bool use_recruit_bias_adjustment =
+                true; /*!< A flag to indicate if recruitment deviations are bias adjusted
              */
-  Type log_sigma_recruit; /*!< Log standard deviation of log recruitment
+        Type log_sigma_recruit; /*!< Log standard deviation of log recruitment
                              deviations */
-  Type rzero; /*!< Unexploited recruitment. Should be a positive value.*/
-  bool estimate_recruit_deviations =
-      true; /*!< A flag to indicate if recruitment deviations are estimated or
+        Type log_rzero; /*!< Unexploited recruitment. Should be a positive value.*/
```

Comment:
suggestion (non-blocking): update documentation to describe that this a log value.